packages/stable/src/__tests__/api/records.int.spec.ts

```diff
@@ -447,4 +447,179 @@ describe('records integration test', () => {
       .sort();
     expect(values).toEqual([1, 2, 3]);
   });
+
+  test('aggregate records with metric aggregates', async () => {
+    const testName = `aggregate_metrics_${randomInt()}`;
+    const source = {
+      type: 'container' as const,
+      space: testSpaceId,
+      externalId: testContainerId,
+    };
+
+    await client.records.ingest(immutableStreamId, [
+      {
+        space: testSpaceId,
+        externalId: `agg_1_${randomInt()}`,
+        sources: [
+          {
+            source,
+            properties: {
+              name: testName,
+              value: 100,
+              timestamp: '2025-01-01T00:00:00.000Z',
+            },
+          },
+        ],
+      },
+      {
+        space: testSpaceId,
+        externalId: `agg_2_${randomInt()}`,
+        sources: [
+          {
+            source,
+            properties: {
+              name: testName,
+              value: 200,
+              timestamp: '2025-01-01T00:00:00.000Z',
+            },
+          },
+        ],
+      },
+    ]);
+
+    await vi.waitFor(
+      async () => {
+        const prop = [testSpaceId, testContainerId, 'value'] as [
+          string,
+          string,
+          string,
+        ];
+        const aggregates = await client.records.aggregate(immutableStreamId, {
+          lastUpdatedTime: {
+            gte: new Date(Date.now() - 24 * 60 * 60 * 1000).toISOString(),
+          },
+          filter: {
+            equals: {
+              property: [testSpaceId, testContainerId, 'name'],
+              value: testName,
+            },
+          },
+          aggregates: {
+            total_count: { count: {} },
+            total_sum: { sum: { property: prop } },
+            avg_value: { avg: { property: prop } },
+            min_value: { min: { property: prop } },
+            max_value: { max: { property: prop } },
+          },
+        });
+        expect((aggregates.total_count as { count: number }).count).toBe(2);
+        expect((aggregates.total_sum as { sum: number }).sum).toBe(300);
+        expect((aggregates.avg_value as { avg: number }).avg).toBe(150);
+        expect((aggregates.min_value as { min: number }).min).toBe(100);
+        expect((aggregates.max_value as { max: number }).max).toBe(200);
```

Comment:
Should we have a discriminator for these types 🤔 ?
https://dev.to/darkmavis1980/what-are-typescript-discriminated-unions-5hbb